pytorch_lightning/core/lightning.py

```diff
@@ -1159,7 +1159,14 @@ def optimizer_step(self, current_epoch, batch_idx, optimizer,
         elif isinstance(optimizer, torch.optim.LBFGS):
             optimizer.step(second_order_closure)
         else:
-            optimizer.step()
+            if self.use_amp and self.use_native_amp:
+                self.trainer.scaler.step(optimizer)
```

Comment:
@mcarilli what happens with an optimizer that needs a second_order_closure?
@soumith

Reply:
Closure use [isn't supported](https://pytorch.org/docs/master/amp.html#torch.cuda.amp.GradScaler.step) right now.  Making closures with dynamic gradient scaling is tricky, but not impossible.  I'm waiting for user demand before I decide if and how to implement it, so if people start asking for it, please file a Pytorch issue and assign mcarilli or add module label "amp (automated mixed precision)"